pkg/uninstaller/deletion.go

```diff
@@ -25,7 +25,7 @@ func UnjoinClusters(ctx context.Context, client dynamic.Interface) error {
 	klog.Infof("Unjoin %v ForeignClusters", len(foreign.Items))
 	r1 := client.Resource(discoveryV1alpha1.ForeignClusterGroupVersionResource)
 	for index := range foreign.Items {
-		patch := []byte(`{"spec": {"join": false}}`)
+		patch := []byte(`{"spec": {"outgoingPeeringEnabled": "No"}}`)
```

Comment:
If possible, it would probably better to avoid manually forging the json patch, as lacking any type check to prevent the occurrence of unnoticed issues in case the API definition gets updated.